### ISSUE TITLE
optimize visual  effects with LOD

### ISSUE THREAD

**Issue by pingfengafei** [va3c/viewer#10]
followed with revit.html. Here I want to solve the problem:
optimizing revit.js file with LOD(leveal of details) for better visual effects.
I noticed that a revit-file.js about 150M size. while I rotate the 3D building，only 12-15fps(under ATI 4650).
I also see that three.js has LOD fucntion.but how can I add it to optimize visual effects??

**Comment by mattmas** [comment]
What I always wanted to see was whether we could split the JSON into multiple streams – identify certain categories that might only be shown at “medium LOD” or “high LOD”.

For example:

Walls, Windows, Topography, etc – These should be shown at all levels of detail.

Furniture, Casework, Electrical Fixtures, Railings – These might be shown at medium detail.

Curtain Wall Mullions, Certain MEP Categories – might only be shown at high-detail.

(the list above is based on my impression of both: what makes sense when you are “zoomed out” on a building, as well as the quantity/complexity of element geometry).

-Matt

**Comment by pingfengafei** [comment]
I agree with your opinion. Identifying different materials with different LOD is a useful way.
But question comes:
Can Revit API provides a convenient way to identify categorie of an objects belonging to？
I know nothing with architecture in real world.

**Comment by jeremytammik** [comment]
yes, absolutely! revit provides exactly the category classification that matt lists above, e.g. walls, windows, furniture, mullions etc.
